spot_driver_cpp/include/spot_driver_cpp/spot_image_publisher.hpp

```diff
@@ -94,10 +94,12 @@ class SpotImagePublisher {
 
   // Interface classes to interact with Spot and the middleware.
   std::unique_ptr<TimerInterfaceBase> timer_interface_;
-  std::unique_ptr<SpotInterfaceBase> spot_interface_;
+  std::unique_ptr<ImageApi> image_api_;
   std::unique_ptr<PublisherInterfaceBase> publisher_interface_;
   std::unique_ptr<ParameterInterfaceBase> parameter_interface_;
   std::unique_ptr<TfInterfaceBase> tf_interface_;
   std::unique_ptr<LoggerInterfaceBase> logger_interface_;
+
+  std::shared_ptr<Robot> robot_;
```

Comment:
Should we consider passing ownership to the image_api? In conjunction with refactoring the use of `spot_image_sources` to the `image_api`? That way the publisher could be "Robot agnostic" and only need to interface with the `image_api`.

@jschornak-bdai thoughts?

Reply:
Because the Robot is used to retrieve the different clients, the main idea is to use it as a shared resourced that can be passed to different APIs, including the reverse kinematic API.

Reply:
We would only need to share the same instance of Robot between different interface implementations if we were creating a single executable which was an agglomeration of different Spot API clients. There's an important design question here -- will we be providing a single executable that exposes all the Spot API clients, or a collection of executables that each expose an individual client or closely-related set of clients?

My preference is to do the latter -- I do not think there is negative overhead in authenticating to Spot from multiple processes in parallel, and we can implement the nodes as component nodes in a container to reduce the ROS 2 middleware overhead which we'd otherwise incurred by having multiple DDS participants from multiple standalone nodes.